rwf/src/controller/auth.rs

```diff
@@ -52,9 +52,9 @@ pub trait Authentication: Sync + Send {
     async fn authorize(&self, request: &Request) -> Result<bool, Error>;
 
     /// If the request is denied, return a specific response.
-    /// Default is `403 - Forbidden`.
+    /// Default is `401 - Unauthorized`.
     async fn denied(&self, request: &Request) -> Result<Response, Error> {
-        Ok(Response::forbidden())
+        Ok(Response::unauthorized("Basic"))
```

Comment:
If we are to return `Basic` by default here, maybe we ought to let the user customize the authentication mechanism name, by introducing another method in the interface perhaps.

Reply:
Yeah, I wasn't too sure which challenge to put here. I can always make the `WWW-Authenticate` header optional, and just leave it out in this case. Given this is the default, I didn't think the user would need to provide it. They can always override it to return their custom response details instead.